src/makefile

```diff
@@ -25,6 +26,13 @@ SRC = $(wildcard *.cpp)\
 			$(wildcard ./structures/*.cpp)\
 			$(wildcard ./utils/*.cpp)
 
+# Remove routing if we don't require it
+ifeq ($(USE_ROUTING),false)
+	SRC := $(filter-out $(wildcard ./routing/*.cpp), $(SRC))
+else
+	LDLIBS += -lssl -lcrypto
+endif
+
 # Checking for libosrm
```

Comment:
Update: checking whether `libosrm` is installed should only happen with `USE_ROUTING=true`. Disabling routing will remove the libosrm includes anyway so we'd then be linking a lot of unnecessary stuff.